src/nanoarrow/utils_inline.h

```diff
@@ -26,6 +26,115 @@
 extern "C" {
 #endif
 
+static inline void ArrowLayoutInit(struct ArrowLayout* layout,
```

Comment:
Just an observation, but I think this sort of function doesn't necessarily have to be inline - it shouldn't be called frequently and shouldn't be a bottleneck

Reply:
Or, of course, you could go the other way and put _everything_ in headers, making this a header-only library